server/server.js

```diff
@@ -85,27 +85,43 @@ app.post(`${apiBaseUrl}/form`,
     return res.json({ id, isolationPlanStatus });
   }));
 
-// Create new form, not secured
+// Submission from PHAC, secured by JWT for PHAC users only
 app.post(`${apiBaseUrl}/phac/submission`,
   passport.authenticate('jwt-phac', { session: false }),
   asyncMiddleware(async (req, res) => {
     await validate(PhacSchema, req.body); // Validate submitted submissions against schema
     const phacCollection = dbClient.db.collection(collections.PHAC);
-    const id = await generateUniqueHexId(phacCollection);
 
     const currentIsoDate = new Date().toISOString();
-    const phacItems = req.body.map((item) => ({
+    const phacItems = await Promise.all(req.body.map(async (item) => ({
       ...item,
-      id,
+      id: await generateUniqueHexId(phacCollection),
       createdAt: currentIsoDate,
       updatedAt: currentIsoDate,
-    }));
-
-    await phacCollection.insertMany(phacItems);
-
-    const idMap = phacItems.reduce((a, v) => ({ ...a, [v.covid_id]: v.id }), {});
+    })));
+
+    const results = { successful: {}, duplicates: [], errors: [] };
+    try {
+      await phacCollection.insertMany(phacItems, { ordered: false });
+      results.successful = phacItems.reduce((a, v) => ({ ...a, [v.covid_id]: v.id }), {});
+    } catch (error) {
+      try {
+        const { writeErrors } = error.result.result;
+        results.duplicates = writeErrors
```

Comment:
Am I missing something? The logic for identifying duplicates vs. errors looks identical to me....

Reply:
Ah lol the code returned by the MongoDB driver is `11000` for duplicates. So it's `===` for dupes and `!==` for everything else (which we consider errors).

Reply:
Ahhh that makes sense, thanks